_episodes_rmd/11-vector-raster-integration.Rmd

```diff
@@ -185,102 +185,92 @@ vegetation plot locations plotted on top of the Canopy Height Model information.
+CHM_plots_HARVcrop_sp_rand_sample = sample_n(CHM_plots_HARVcrop_sp, 10000)
+```
 
+```{r, echo = FALSE}
+ggplot() + 
+  geom_sf(data = st_convex_hull(st_union(CHM_rand_sample)), fill = "green") +
+  geom_sf(data = st_convex_hull(st_union(lines_HARV)), fill = "purple", aes(alpha = 0.1)) +
+  geom_sf(data = st_convex_hull(st_union(CHM_plots_HARVcrop_sp_rand_sample)), fill = "darkgreen") +
+  geom_sf(data = st_convex_hull(st_union(plot_locations_sp_HARV)), fill = "black") + 
+  geom_sf(data = aoi_boundary_HARV, fill = "blue") +
+  theme(legend.position = "none")
 ```
 
 ## Define an Extent
-We can also use an `extent()` method to define an extent to be used as a cropping
-boundary. This creates an object of class `extent`.
+So far, we have used a shapefile to crop the extent of a raster dataset. Alternatively,
+we can also the
+`extent()` function to define an extent to be used as a cropping
+boundary. This creates a new object of class `Extent`. Here we will 
+provide the `extent()`
+function our xmin, xmax, ymin, and ymax (in that order). 
 
-```{r hidden-extent-chunk}
-# extent format (xmin, xmax, ymin, ymax)
+```{r}
 new.extent <- extent(732161.2, 732238.7, 4713249, 4713333)
 class(new.extent)
 ```
 
-Once we have defined the extent, we can use the `crop` function to crop our
-raster.
+> ## Data Tip
+> The `extent()` function can be created from a numeric vector (as shown above),
+> a matrix, or a list. For more details see the `extent()` function help file. 
+> * `??raster::extent`
+> * More on the [extent class in `R`](http://www.inside-r.org/packages/cran/raster/docs/extent).
+{: .callout}
 
-```{r crop-using-drawn-extent}
+Once we have defined our new extent, we can use the `crop()` function to crop our
+raster to this extent object.
 
-# crop raster
-CHM_HARV_manualCrop <- crop(x = chm_HARV, y = new.extent)
-
-# plot extent boundary and newly cropped raster
-plot(aoi_boundary_HARV$geometry,
-     main = "Manually Cropped Raster\n NEON Harvard Forest Field Site")
-plot(new.extent,
-     col = "brown",
-     lwd = 4,
-     add = TRUE)
-plot(CHM_HARV_manualCrop,
-     add = TRUE)
+```{r crop-using-drawn-extent}
+CHM_HARV_manualCrop <- crop(x = CHM_HARV, y = new.extent)
 ```
 
-Notice that our manual `new.extent` (in red) is smaller than the
-`aoi_boundary_HARV` and that the raster is now the same as the `new.extent`
-object.
+To plot this data using `ggplot()` we need to convert it to a dataframe. 
+
+```{r}
+CHM_HARV_manualCrop_df <- raster::as.data.frame(CHM_HARV_manualCrop, xy = TRUE)
+```
 
-See the documentation for the `extent()` function for more ways
-to create an `extent` object.
+Now we can plot this cropped data. We will show the AOI boundary on the same plot for scale.
 
-* `??raster::extent`
-* More on the
-<a href="http://www.inside-r.org/packages/cran/raster/docs/extent" target="_blank">
-extent class in `R`</a>.
+```{r}
+ggplot() + 
+    geom_sf(data = aoi_boundary_HARV, color = "blue") +
```

Comment:
if you add `fill = NA` to the first call - `geom_sf(data = aoi_boundary_HARV, color = "blue") +` you can then move the call so it draws after/over the raster. That way the outline doesn't get cut off

Reply:
Great, thanks! I was trying to figure out how to do this and having a heckofa time.